services/document-manager/backend/app/docman/utils/document_upload_helper.py

```diff
@@ -285,3 +293,153 @@ def complete_upload(cls, key, new_key, doc_guid, versions=None, version_guid=Non
+                raise ValueError(
+                    f'Found non spatial bundle file types in spatial bundle: {", ".join(extra_extensions)}')
+
+    @classmethod
+    def zip_spatial_files(cls, bundle_documents, name):
+        oss_service = ObjectStoreStorageService()
+        shpz_path = secure_filename(f'{name}.shpz')
+
+        with zipfile.ZipFile(shpz_path, 'w', zipfile.ZIP_DEFLATED) as zipf:
+            for doc in bundle_documents:
+                response = oss_service.download_file(doc.object_store_path, doc.file_display_name, False)
+
+                if response.status_code == 200:
+                    current_app.logger.info(f"Successfully downloaded document: {doc.file_display_name}")
+                    file_data = response.get_data()
+
+                    zipf.writestr(doc.file_display_name, file_data)
+                else:
+                    raise Exception(f"Failed to download document: {doc.file_display_name}")
+
+        return shpz_path
+
+    @classmethod
+    def download_kml_kmz_files(cls, doc):
+        # Download the file from object store
+        response = ObjectStoreStorageService().download_file(doc.object_store_path, doc.file_display_name, False)
+
+        # Check the response status
+        if response.status_code == 200:
+            file_data = response.get_data()  # obtain file data from response body
+
+            dir_path = 'tmp/spatial'
+            os.makedirs(dir_path, exist_ok=True)
+            file_path = os.path.join(dir_path, doc.file_display_name)
+            with open(file_path, 'wb') as file:
+                file.write(file_data)
+        else:
+            raise FileNotFoundError(f"Failed to download file: {doc.file_display_name}")
+
+    @classmethod
+    def send_spatial_file_to_geomark(cls, file_path):
+        file_extension = os.path.splitext(file_path)[1].lstrip('.')
+
+        # Ensure the format is one of the allowed ones
+        assert file_extension in ['shpz', 'kml', 'kmz'], "Invalid file format. Must be one of: 'shpz', 'kml', 'kmz'."
+
+        mime_type = mimetypes.guess_type(file_path)[0]
+
+        url = 'https://apps.gov.bc.ca/pub/geomark/geomarks/new'
+
+        try:
+            with open(file_path, 'rb') as file:
+                multipart_data = MultipartEncoder(
+                    fields={
+                        'body': (os.path.basename(file_path), file, mime_type),
+                        'resultFormat': 'json',
+                        'format': file_extension,
+                        'srid': '4326'
+                    }
+                )
+                headers = {
+                    'Content-Type': multipart_data.content_type
+                }
+                # Send the request
+                response = requests.post(
+                    url,
+                    data=multipart_data,
+                    headers=headers
+                )
+        finally:
+            # Delete the file
+            os.remove(file_path)
```

Comment:
I think we'll have the same issue here with a storage leak that the downloaded file will still be present if the `assert` above fails.

Possible to wrap this whole method to be safe?